modules/reactor/include/meshgenerators/HexagonMeshTrimmer.h

```diff
@@ -0,0 +1,162 @@
+                   const std::vector<Real> bdry_pars,
+                   const subdomain_id_type block_id_to_remove,
+                   const std::set<subdomain_id_type> subdomain_ids_set,
+                   const boundary_id_type trimming_section_boundary_id,
+                   const boundary_id_type external_boundary_id = OUTER_SIDESET_ID,
+                   const bool side_to_move = true);
+
+  /**
+   * Determines whether a point on XY-plane is on the side of a given line that needs to be removed
+   * @param px x coordinate of the point
+   * @param py y coordinate of the point
+   * @param param_1 parameter 1 (a) in line formula a*x+b*y+c=0
+   * @param param_2 parameter 2 (b) in line formula a*x+b*y+c=0
+   * @param param_3 parameter 3 (c) in line formula a*x+b*y+c=0
+   * @param direction_param which side is the side that needs to be removed
+   * @param dis_tol tolerance used in determining side
+   * @return whether the point is on the side of the line that needed to be removed
+   */
+  bool lineSideDeterminator(const Real px,
+                            const Real py,
+                            const Real param_1,
+                            const Real param_2,
+                            const Real param_3,
+                            const bool direction_param,
+                            const Real dis_tol = 1.0E-6);
+
+  /**
+   * Calculates the intersect Point of two given straight lines
+   * @param param_11 parameter 1 (a) in line formula a*x+b*y+c=0 for the first line
+   * @param param_12 parameter 2 (b) in line formula a*x+b*y+c=0 for the first line
+   * @param param_13 parameter 3 (c) in line formula a*x+b*y+c=0 for the first line
+   * @param param_21 parameter 1 (a) in line formula a*x+b*y+c=0 for the second line
+   * @param param_22 parameter 2 (b) in line formula a*x+b*y+c=0 for the second line
+   * @param param_23 parameter 3 (c) in line formula a*x+b*y+c=0 for the second line
+   * @return intersect point of the two lines
+   */
+  Point twoLineIntersection(const Real param_11,
+                            const Real param_12,
+                            const Real param_13,
+                            const Real param_21,
+                            const Real param_22,
+                            const Real param_23);
+
+  /**
+   * Fixes degenerate QUAD elements created by the hexagonal mesh trimming by converting them into
+   * TRI elements
+   * @param mesh input mesh with degenerate QUAD elements that need to be fixed
+   * @param subdomain_ids_set all the subdomain ids in the input mesh
+   * @return whether any elements have been fixed
+   */
+  bool quasiTriElementsFixer(ReplicatedMesh & mesh,
+                             const std::set<subdomain_id_type> subdomain_ids_set);
+
+  /**
+   * Calculates the internal angles of a given 2D element
+   * @param elem the element that needs to be investigated
+   * @return sizes of all the internal angles
```

Comment:
```suggestion
   * @return sizes of all the internal angles, sorted by their size
```

I would actually see a sorting by nodes if it were a utility

Reply:
I dont see the need for it to be a util though
do you? do you have further plans with this?

Reply:
I am think about some general trimmer in the future.